flows/boundary.py

```diff
@@ -634,17 +634,66 @@ def get_vespa_passages_from_query_response(
         dig(passage_root, "children", 0, "children", 0)
         for passage_root in passages_root
     ]
-    vespa_passages: dict[TextBlockId, tuple[VespaHitId, VespaPassage]] = {
-        passage["fields"]["text_block_id"]: (
-            passage["id"],
-            VespaPassage.model_validate(passage["fields"]),
+
+    vespa_passages: dict[TextBlockId, tuple[VespaHitId, VespaPassage]] = {}
+    for passage in passage_hits:
+        fields = passage.get("fields")
+        if not fields:
+            error = (
+                f"Vespa passage with no 'fields': {passage}, "
+                f"sample of passage hits: {passage_hits[:5]}"
+            )
+            print(error)
+            raise ValueError(error)
+
+        text_block_id = fields.get("text_block_id")
+        if not text_block_id:
+            error = (
+                f"Vespa passage with no 'text_block_id' in passage: {passage}, "
+                f"sample of passage hits: {passage_hits[:5]}"
+            )
+            print(error)
+            raise ValueError(error)
+
+        text_block_id = TextBlockId(text_block_id)
+        passage_id = VespaHitId(passage.get("id"))
+        vespa_pasage = VespaPassage.model_validate(fields)
+
+        vespa_passages[text_block_id] = (
+            passage_id,
+            vespa_pasage,
         )
-        for passage in passage_hits
-    }
 
     return vespa_passages
 
 
+@vespa_retry(
+    max_attempts=2,
+    exception_types=(
+        ValueError,
+        QueryError,
+    ),
+)
+async def make_query_and_extract_passages(
+    vespa_connection_pool: VespaAsync,
+    query: qb.Query,
+    query_profile: str,
+) -> tuple[VespaQueryResponse, dict[TextBlockId, tuple[VespaHitId, VespaPassage]]]:
+    """Make the query and extract the passages."""
+
+    vespa_query_response: VespaQueryResponse = await vespa_connection_pool.query(
+        yql=query,
+        queryProfile=query_profile,
+    )
+
+    if not vespa_query_response.is_successful():
+        raise QueryError(vespa_query_response.get_status_code())
+
+    return vespa_query_response, get_vespa_passages_from_query_response(
+        vespa_query_response
+    )
```

Comment:
@jesse-c  Thoughts on whether this needs a test? The functionality is essentially tested during the testing of the `get_document_passages_from_vespa__generator` function and testing that we retry seems like it would just be duplicating any tests for the `vespa_retry` function.

Reply:
Lifting out to it's own function so it's explicit that we're decorating with retries.